x-pack/legacy/plugins/apm/public/components/app/Settings/AddSettings/AddSettingFlyoutBody.tsx

```diff
@@ -0,0 +1,270 @@
+        </EuiFormRow>
+
+        <EuiFormRow
+          label={i18n.translate(
+            'xpack.apm.settings.cm.flyOut.serviceEnvironmentSelectLabel',
+            {
+              defaultMessage: 'Environment'
+            }
+          )}
+          helpText={i18n.translate(
+            'xpack.apm.settings.cm.flyOut.serviceEnvironmentSelectHelpText',
+            {
+              defaultMessage:
+                'Only a single environment per configuration is supported.'
+            }
+          )}
+          error={i18n.translate(
+            'xpack.apm.settings.cm.flyOut.serviceEnvironmentSelectErrorText',
+            {
+              defaultMessage:
+                'Must select a valid environement to save a configuration.'
+            }
+          )}
+          isInvalid={
+            !(
+              selectedConfig ||
+              (!selectedConfig &&
+                environment &&
+                isSelectedEnvironmentValid &&
+                environmentStatus === 'success') ||
+              isNaN(sampleRate)
+            )
+          }
+        >
+          <EuiSelect
+            key={serviceName} // rerender when serviceName changes to mitigate initial selection bug in EuiSelect
+            isLoading={environmentStatus === 'loading'}
+            hasNoInitialSelection
+            options={environmentOptions}
+            value={environment}
+            disabled={!serviceName || Boolean(selectedConfig)}
+            onChange={e => {
+              e.preventDefault();
+              setEnvironment(e.target.value);
+            }}
+          />
+        </EuiFormRow>
+
+        <EuiTitle size="xs">
+          <h3>
+            {i18n.translate(
+              'xpack.apm.settings.cm.flyOut.configurationSectionTitle',
+              {
+                defaultMessage: 'Configuration'
+              }
+            )}
+          </h3>
+        </EuiTitle>
+
+        <EuiSpacer size="m" />
+
+        <EuiFormRow
+          label={i18n.translate(
+            'xpack.apm.settings.cm.flyOut.sampleRateConfigurationInputLabel',
+            {
+              defaultMessage: 'Transaction sample rate'
+            }
+          )}
+          helpText={i18n.translate(
+            'xpack.apm.settings.cm.flyOut.sampleRateConfigurationInputHelpText',
+            {
+              defaultMessage:
+                'Choose a rate between 0.00 and 1.0. Default configuration is 1.0 (100% of traces).'
+            }
+          )}
+          error={i18n.translate(
+            'xpack.apm.settings.cm.flyOut.sampleRateConfigurationInputErrorText',
+            {
+              defaultMessage: 'Sample rate must be between 0.00 and 1'
+            }
+          )}
+          isInvalid={
+            !(
+              (Boolean(selectedConfig) &&
+                (isNaN(sampleRate) || isSampleRateValid)) ||
+              (!selectedConfig &&
+                (!(serviceName || environment) ||
+                  (isNaN(sampleRate) || isSampleRateValid)))
```

Comment:
Can this be extracted to something like `shouldShowSampleRateError`?

Reply:
i plan on fixing some of the issues with validation in another issue (https://github.com/elastic/kibana/issues/40114) so i can clean up the ui validation code there as well